bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/factory/HueThingHandlerFactory.java

```diff
@@ -152,9 +172,15 @@ private ThingUID getThingUID(ThingTypeUID thingTypeUID, String id, @Nullable Thi
 
     @Override
     protected @Nullable ThingHandler createHandler(Thing thing) {
-        if (HueBridgeHandler.SUPPORTED_THING_TYPES.contains(thing.getThingTypeUID())) {
-            return new HueBridgeHandler((Bridge) thing, httpClient, stateDescriptionProvider, i18nProvider,
-                    localeProvider);
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+        if (HueBindingConstants.THING_TYPE_BRIDGE_API2.equals(thingTypeUID)) {
+            return new Clip2BridgeHandler((Bridge) thing, httpClientFactory, thingRegistry,
+                    getBundleContext().getBundle(), localeProvider, i18nProvider);
```

Comment:
> Making things consistent would be better.

Understood. I was forced to add a bundle argument because `Clip2BridgeHandler` needs to export a method 'getLocalizedText()` which is used a) in the console command extension, and b) in the `Clip2ThingDiscovery` service.

Reason is that the Hue Bridge has a 'super' zone for all lights in the whole home, and in the bridge that zone name seems to be hardwired in English ('All Lights') whereas we want to offer non English speaking users a friendly name for that.

Reply:
Yes but in class `HueBridgeHandler`, we also gut the bundle. It is just two different ways to do it.
Maybe your option is better.

Reply:
Wait please. If your way is better, we keep it (and we can update the old one).